test/functional/forall/atomic-view/tests/test-forall-AtomicMultiView.hpp

```diff
@@ -0,0 +1,120 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+///
+/// Header file containing basic functional tests for atomic operations with forall and views.
+///
+
+#ifndef __TEST_FORALL_ATOMIC_MULTIVIEW_HPP__
+#define __TEST_FORALL_ATOMIC_MULTIVIEW_HPP__
+
+template <typename ExecPolicy,
+          typename AtomicPolicy,
+          typename WORKINGRES,
+          typename IdxType,
+          typename T>
+void ForallAtomicMultiViewTestImpl( IdxType N )
+{
+  RAJA::TypedRangeSegment<IdxType> seg(0, N);
+  RAJA::TypedRangeSegment<IdxType> seg_hundred(0, 100);
+  RAJA::TypedRangeSegment<IdxType> seg_twohundred(0, 200);
+
+  camp::resources::Resource work_res{WORKINGRES()};
+  camp::resources::Resource host_res{camp::resources::Host()};
+
+  T * actualsource = work_res.allocate<T>(N);
+  T * actualdest = work_res.allocate<T>(N/2);
+  T * check_array = host_res.allocate<T>(N/2);
+
+  T ** source = new T * [200];
+  for ( int ii = 0; ii < 200; ++ii )
+  {
+    source[ii] = actualsource+(ii*200);
```

Comment:
Shouldn't this depend on N somehow instead of always being 200*200 = 40000, especially because N = 20000.

Reply:
@MrBurmark The indices now depend on N. Hopefully it's still relatively clear, let me know what you think.

Reply:
That looks better.